packages/cht/test/Adaptor.test.js

```diff
@@ -0,0 +1,198 @@
+import { expect } from 'chai';
+import { enableMockClient } from '@openfn/language-common/util';
+
+import { request, get, post, put } from '../src/Adaptor.js';
+
+const testServer = enableMockClient('https://example.cht.com');
+
+describe('request', () => {
+  it('should send a GET request', async () => {
+    const data = {
+      has_permissions: true,
+    };
+
+    testServer
+      .intercept({
+        path: '/api/v1/settings',
+        method: 'GET',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, data);
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+      data,
+    };
+
+    const finalState = await request('GET', '/api/v1/settings')(state);
+
+    expect(finalState.data).to.eql(data);
+  });
+
+  it('should throw for a 403 response', async () => {
+    testServer
+      .intercept({
+        path: '/api/v1/contacts',
+        method: 'GET',
+      })
+      .reply(403);
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+    };
+
+    const error = await request(
+      'GET',
+      '/api/v1/contacts'
+    )(state).catch(error => {
+      return error;
+    });
+    expect(error.statusMessage).to.eql('Forbidden');
+  });
+
+  it('should include basic auth', async () => {
+    testServer
+      .intercept({
+        path: '/api/v1/settings',
+        method: 'GET',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, {});
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+      data: {},
+    };
+
+    const finalState = await request('GET', '/api/v1/settings')(state);
+
+    expect(finalState.configuration).to.eql(state.configuration);
+  });
+});
+
+describe('get()', () => {
+  it('should make a simple get request', async () => {
```

Comment:
I would actually like to see at least one test which includes query parameters.

You can either extend this test or add a second one like `should make a get with parameters`